File: EntityFX.NetBenchamarks.Core/Generic/StringManipulationBase.cs
using System.Runtime.CompilerServices;

namespace EntityFX.NetBenchmark.Core.Generic
{
    public abstract class StringManipulationBase<TResult> : BenchmarkBase<TResult>
    {
        protected double R;

        public StringManipulationBase()
        {
            Iterrations = 5000000;
            Ratio = 2;
        }

        [MethodImpl(MethodImplOptions.AggressiveInlining)]
        protected static string DoStringManipilation(string str)
        {
            return (string.Join("/", str.Split(' ')).Replace("/", "_").ToUpper() + "AAA").ToLower().Replace("aaa", ".");
        }
    }
}

File: src/python/entityfx/string_manipulation_base.py

from entityfx.benchmark_base import BenchmarkBase
from entityfx.writer import Writer
import math, time

class StringManipulationBase(BenchmarkBase):
    
    def __init__(self, writer: Writer, print_to_console : bool=True, is_enabled : bool=True) -> None:
        super().__init__(writer, print_to_console, is_enabled)
        self._iterrations = 5000000
        self.ratio = (10)
    
    @staticmethod
    def _doStringManipilation(str0_ : str) -> str:
        return ("/".join(str0_.split(' ')).replace("/", "_").upper() + "AAA").lower().replace("aaa", ".")